x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -698,4 +701,47 @@ export class FleetPlugin
 
     return this.logger;
   }
+
+  private async initializeUninstallTokens() {
+    await this.generateUninstallTokens();
+    this.validateUninstallTokens();
+  }
+
+  private async generateUninstallTokens() {
+    const logger = appContextService.getLogger();
+
+    logger.debug('Generating Agent uninstall tokens');
+    if (!appContextService.getEncryptedSavedObjectsSetup()?.canEncrypt) {
+      logger.warn(
+        'xpack.encryptedSavedObjects.encryptionKey is not configured, agent uninstall tokens are being stored in plain text'
+      );
+    }
+    await appContextService.getUninstallTokenService()?.generateTokensForAllPolicies();
+
+    if (appContextService.getEncryptedSavedObjectsSetup()?.canEncrypt) {
+      logger.debug('Checking for and encrypting plain text uninstall tokens');
+      await appContextService.getUninstallTokenService()?.encryptTokens();
```

Comment:
not a big speed gain but we can probably leave this async since nothing depends on it being encrypted.

Reply:
I left the `await` here, so every task that possibly modify tokens (generation and encryption) finishes first, before validation starts on the same dataset, so we validate the final results

the Fleet plugin start does not wait for any of these sequential steps, though